platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationConfigInstaller.java

```diff
@@ -138,6 +118,37 @@ public void run() {
     }
   }
 
+  private void startApp(String appName, String appLocation) {
+    try {
+      if (!appLocation.isEmpty()) {
+        appService.addApplication(new URI(appLocation));
+      }
+      executeAsSystem(
+          () -> {
+            appService.startApplication(appName);
+            return true;
+          });
+    } catch (ApplicationServiceException ase) {
+      LOGGER.warn("Could not start " + appName, ase);
+    } catch (URISyntaxException use) {
+      LOGGER.warn("Could not install application, location is not a valid URI " + appLocation, use);
+    }
+  }
+
+  private void installAndUninstallFeatures() {
+    try {
+      if (StringUtils.isNotBlank(postInstallFeatureStart)) {
+        featuresService.installFeature(
+            postInstallFeatureStart, EnumSet.of(FeaturesService.Option.NoAutoRefreshBundles));
+      }
+      if (StringUtils.isNotBlank(postInstallFeatureStop)) {
+        featuresService.uninstallFeature(postInstallFeatureStop);
+      }
+    } catch (Exception e) {
```

Comment:
Is there a more specific exception we can catch here?

Reply:
Unfortunately, not with Karaf. They declare they throw Exception 😞